File: eCinema.Web.API/ClassLibrary1/Entities/Reservation.cs
using eCInema.Models.Dtos.Customer;
using eCInema.Models.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCInema.Models.Entities
{
    public class Reservation
    {
        public  int Id { get; set; }
        public Customer Customer{ get; set; }
        public int CustomerId { get; set; }
        public int ScheduleId { get; set; }
        public Schedule Schedule { get; set; }
        public int NumberOfTickets { get; set; }
        public double Price { get; set; }
        public List<ScheduleSeat> ScheduleSeat { get; set; }
        public ReservationStatusEnum Status { get; set; }

    }
}

File: eCinema.Web.API/ecinemamobile/lib/models/Reservations/reservation.dart
import 'package:ecinemamobile/models/Halls/hall.seats.dart';
import 'package:ecinemamobile/models/Reservations/reservation.status.dart';
import '../ScheduleSeats/schedule.seat.dart';
import '../Schedules/schedule.dart';
part 'reservation.g.dart';

class Reservation {
  int? scheduleId;
  int? numberOfTickets;
  List<ScheduleSeat>? scheduleSeats;
  double? price;
  ReservationStatusEnum? status;
  String? paymentID;
  Reservation(this.scheduleId, this.numberOfTickets, this.scheduleSeats,
      this.price, this.status,
      [this.paymentID]);
  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
